### DISCUSSION TITLE
Possible bug with Obsidian 1.6.3, BRAVE and Web Clipper API

### DISCUSSION THREAD

**Discussion by Penworks** [coddingtonbear/obsidian-web#200] [Exotic Browser Support] [answered]
### Describe the bug

Ive been using web clipper for at least 18 months, no problem, its very useful. I use on a Win 10 PC and on and a High Sierra MBPro14. The PC runs Obsidian 1.6.3, the Mac is v1.5.12. Im now getting errors of "Unable to connect to Obsidian: TypeError: Failed to fetch...." on the Win PC. 

I use Brave browser on both units, with shields up. 

Brave version on the HS Mac is older, 1.57.64, obsidian clipper still works fine.

Brave on the PC is currently 1.67.116, Obsidian clipper cannot fetch...

Edit - after some other testing with Chrome and Obsidian 1.6.3 it seems that it is Brave that is the culprit.

I have updated the API key. Still broken.

I have followed the procedure in [this post](https://github.com/coddingtonbear/obsidian-web/wiki/How-do-I-get-my-browser-trust-my-Obsidian-Local-REST-API-certificate%3F#managing-previous-certificates), also rebooted the PC, but it hasn't made any difference.

![image](https://github.com/coddingtonbear/obsidian-web/assets/758626/2a735c7e-256c-47f0-a737-c6c0abe1352d)

### To Reproduce

To reproduce you can set versions of Brave and of Obsidian version to be the same as above, and compare.  Then test with Chrome or other browsers to see its still working, to confirm its Brave.

### Expected Behavior

API will fail on updated Brave, even with cert downloaded and installed in Win OS Trusted Auth

### Operating System

Win 10

### Browser

Brave

### Browser Version

1.67.116

**Comment by Penworks** [comment, marked as answer]
Update. 
I managed to solve this problem so it isnt a bug its an issue that *might* happen if using Brave. I did the following:

- Refreshed the API key
- downloaded a new certificate and installed in Local Machine (not only at user level) , in Trusted Root Certification Authorities
- reinstalled Brave and enabled a few localhost permissions in https://brave/flags (hard to say if this made any difference)
- rebooted the PC again

This restored functionality for the web clipper. 

If I could remove the original post I probably would but I cant. It might be useful for others and I can move it to another category of discussion.

**Reply by Mladen-Rusev** [reply]
This solved it for me on Brave, Windows 10. For those wondering step 2 is the important one. I did not have to reboot PC or enable any flags.
> downloaded a new certificate and installed in Local Machine (not only at user level) , in Trusted Root Certification Authorities

I clicked on "Automatically select the certificate store based on the type of cert" on my first attempt and it didn't work.
Instead you have to choose "Place cert in the following store" and select `Trusted Root Certification Authorities`

![image](https://github.com/user-attachments/assets/bccfaaa9-81a5-4a28-a7e9-bd05b4a1a731)

Thanks to @Penworks for solving this 🙇

**Comment by iwyatt** [comment]
I found Brave has brave://settings/certificates where I imported the certificate I generated in the plugin's settings from Obsidian.
Than I noticed under the authorities tab obsidian web was listed but was not trusted, so I changed the flags to trust it and then it worked. 
I tested that this worked for my system. 

Before the above steps, I also re-added the certificate on my system (Linux Mint) by following these directions:
> 1. Copy the certificate file to /usr/local/share/ca-certificates/: sudo cp your_cert.crt /usr/local/share/ca-certificates/your_cert.crt
> 2. Update the CA store: sudo update-ca-certificates

**Comment by Penworks** [comment]
So its now Nov 2025, my (Win 10) Brave just auto updated and again the clipper is broken. I have tried numerous times to get this working. Installed deleted changed certificates, installed them as trusted in Brave etc etc. Nothing works. On an old version of Brave on my MBPro 2014, it works fine, but I do run it on insecure, to make it very versatile.
